tests/test_node_monitor.py

```diff
@@ -92,27 +101,74 @@ def test_control():
     nm.broadcast_alerts()
     assert mock_email_bot.send_emails.call_count == 0
     assert mock_slack_bot.send_message.call_count == 0
-    mock_email_bot.reset_mock()
+    assert mock_telegram_bot.send_message.call_count == 0
     mock_slack_bot.reset_mock()
+    mock_email_bot.reset_mock()
+    mock_telegram_bot.reset_mock()
     mock_node_provider_db.reset_mock()
 
     # test broadcast_status_report()
     nm.broadcast_status_report()
     assert mock_email_bot.send_emails.call_count == 1
     assert mock_slack_bot.send_message.call_count == 1
+    assert mock_telegram_bot.send_message.call_count == 1
+    mock_slack_bot.reset_mock()
+    mock_email_bot.reset_mock()
+    mock_telegram_bot.reset_mock()
+    mock_node_provider_db.reset_mock()
+
+
+
+def test_control_only_email_bot():
+    """Test the control case. No nodes down. Initialize with only email"""
+    # init
+    mock_email_bot = Mock(spec=EmailBot)
+    mock_slack_bot = Mock(spec=SlackBot)
+    mock_telegram_bot = Mock(spec=TelegramBot)
+    nm = NodeMonitor(mock_node_provider_db, mock_email_bot, 
+                     None, None)
+    nm._resync(cached['control'])
+    nm._resync(cached['control'])
+    nm._resync(cached['control'])
+
+    # test _analyze()
+    nm._analyze()
+    assert len(nm.compromised_nodes) == 0
+    assert len(nm.compromised_nodes_by_provider) == 0
+    assert len(nm.actionables) == 0
+
+    # test broadcast_alerts()
+    nm.broadcast_alerts()
+    assert mock_email_bot.send_emails.call_count == 0
+    assert mock_slack_bot.send_message.call_count == 0
+    assert mock_telegram_bot.send_message.call_count == 0
+    mock_slack_bot.reset_mock()
+    mock_email_bot.reset_mock()
+    mock_telegram_bot.reset_mock()
+    mock_node_provider_db.reset_mock()
+
+    # test broadcast_status_report()
+    nm.broadcast_status_report()
+    assert mock_email_bot.send_emails.call_count == 1
+    assert mock_slack_bot.send_message.call_count == 0
+    assert mock_telegram_bot.send_message.call_count == 0
     mock_slack_bot.reset_mock()
     mock_email_bot.reset_mock()
+    mock_telegram_bot.reset_mock()
     mock_node_provider_db.reset_mock()
```

Comment:
I'm guessing the idea here was to make sure it still works when `slack_bot` and `telegram_bot` are both `None`?

I'm not sure we need this. I think it adds code bloat and is too similar to test above. `NodeMonitor` already has an if-statement within, to validate both of these attributes are added to the class. Thoughts?

Reply:
> I'm guessing the idea here was to make sure it still works when `slack_bot` and `telegram_bot` are both `None`?

Yes!

> I'm not sure we need this. I think it adds code bloat and is too similar to test above. NodeMonitor already has an if-statement within, to validate both of these attributes are added to the class. Thoughts?

I see what you mean, but without these tests, we cannot really verify that Node Monitor runs correctly when Slack and Telegram arguments are `None`.  So I think they should stay.